### DISCUSSION TITLE
[API] Nested filtering

### DISCUSSION THREAD

**Discussion by maximilianofried** [strapi/strapi#7596] [Legacy Topics]
Hello i want to know what is the way to do an api request using nested filter.

For example my Article object look like this:

`[
    {
        "status": "published",
        "country": "germany",
        "categories": [
            {
                "_id": "5f3a57a84916ca0337385653",
                "name": "Start page",
                "createdAt": "2020-08-17T10:10:48.135Z",
                "updatedAt": "2020-08-17T10:10:48.135Z",
                "__v": 0,
                "id": "5f3a57a84916ca0337385653"
            },
            {
                "_id": "5f2930774916ca0337385630",
                "name": "Science",
                "createdAt": "2020-08-04T09:55:03.068Z",
                "updatedAt": "2020-08-20T13:59:03.455Z",
                "__v": 0,
                "link": "/science",
                "id": "5f2930774916ca0337385630"
            }
]`

How can i build my url? Somethink like

http://myurl.com/articles?categories[name]=Art

**Comment by derrickmehaffy** [comment]
Hi @maximilianofried yes we do support deep filtering (though usage of it is advised against as it doesn't have very good performance at the moment)

http://myurl.com/articles?categories.name=Art

**Reply by maximilianofried** [reply]
Thank you Derrick! I have tried that way but I get 500 error. The only thing that worked was only for the "_id", and using _where:

http://myurl.com/articles?_where[0][categories][_id]=5f2930814916ca0337385632

But i can't make it work for "name".

**Reply by derrickmehaffy** [reply]
You are on the v3.1.x Strapi version correct? The complex filtering is new

**Reply by maximiliano-gt** [reply]
I was not. Now I have migrated everything to the new version and the scenario is the following:

-  In development mode is working with this URL -> http://myurl.com/articles?categories.name=Art

-  In production mode, I get this mongo error: 

`MongoError: arguments to $lookup must be strings, let: { localId: "$_id", localAlias: "$main_image" } is type object
`

Do you know with which Mongo version is this feature compatible?

**Reply by derrickmehaffy** [reply]
You need MongoDB v3.6 or higher for Strapi https://github.com/strapi/strapi#-requirements

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.
